sonar-scanner-engine/src/main/java/org/sonar/scanner/report/ComponentsPublisher.java

```diff
@@ -167,6 +169,7 @@ private boolean shouldSkipComponent(DefaultInputComponent component, Collection<
   private void writeVersion(DefaultInputModule module, ScannerReport.Component.Builder builder) {
     String version = getVersion(module);
     if (version != null) {
+      checkArgument(version.length() <= 100, "The version name must not be longer then 100 characters");
```

Comment:
@henryju Could you please have a look at this? Is this the way to solve this? Or would it be better here: https://github.com/SonarSource/sonarqube/blob/master/sonar-plugin-api/src/main/java/org/sonar/api/batch/bootstrap/ProjectDefinition.java#L194 ?

Reply:
@bartfastiel we usually try to fail as soon as possible. I suggest to do the check in ProjectReactorValidator

Reply:
Also note that it would be safer to also validate on CE side, when reading the field from the report, since on SonarCloud we may receive corrupted report (using a modified scanner for example).